_posts/2024-02-01-wildfly-glow.adoc

```diff
@@ -0,0 +1,504 @@
+contained in WildFly layers is documented in this http://docs.wildfly.org/wildfly-galleon-feature-packs/[documentation].
+
+When enabling verbose mode (`--verbose` option with WildFly Glow CLI, `<verbose>true</verbose>` option in the maven plugin), 
+WildFly Glow will output for each layer the rules that matched.
+
+An example of output:
+
+----
+...
+layers inclusion rules
+* ee-core-profile-server
+  - BASE_LAYER
+* ee-concurrency
+  - JAVA_TYPE: [jakarta.enterprise.concurrent.*]
+* undertow-https
+  - ADD_ON
+...
+----
+
+This output means:
+
+* `ee-core-profile-server` is a base layer (always included).
+* `ee-concurrency` layer is included because a Java class located in the java package `jakarta.enterprise.concurrent` has been found.
+* `undertow-https` is included because it is bound to an included add-on (in this case `ssl`).
+
+== Trimming numbers with WildFly Glow
+
+We have measured the Disk usage and Memory consumption of some WildFly quickstarts 
+(using XXXTODOLINK WildFly Glow branch vs not using XXXTODOLINK WildFly Glow branch). We have observed a reduction of 5% to 55% for disk usage 
+and 5% to 32% for memory consumption. Variation bound to the complexity of the quickstart. If the quickstart requires all the Galleon layers 
+present in the aggegator layer (eg: `cloud-server` or `jaxrs-server`) then the gain is lower.
+
+My colleague Kabir Khan has wrote an interesting https://github.com/kabir/vlog-glow/blob/main/README.md[project] 
+and associated Vlog that showcases the gain you can expect with WildFly Glow. If you are interested by WildFly Glow 
+you should really follow his Vlog, it is a very nice video on the topic.
+
+== Datasources support, missing datasource detection
+
+WildFly Glow detects that your deployment uses datasources and will abort asking you to take an action to fix the problem.
+It will suggest the set of known add-ons allowing to connect to database.
+
+You will have to choose one of the proposed add-ons. 
+
+An example of reported error when using the WildFly Glow CLI:
+
+[source,bash]
+----
+wildfly-glow scan todo-backend.war
+
+Wildfly Glow is scanning...
+context: bare-metal
+enabled profile: none
+galleon discovery
+- feature-packs
+   org.wildfly:wildfly-galleon-pack:30.0.1.Final
+- layers
+   ee-core-profile-server
+   ejb-lite
+   jpa
+   jaxrs
+
+identified errors
+* unbound datasources error: java:jboss/datasources/ToDos
+  To correct this error, enable one of the following add-ons:
+  - mariadb
+  - mssqlserver
+  - mysql
+  - oracle
+  - postgresql
+
+Some suggestions have been found. You could enable suggestions with --suggest option (if using the WildFly Glow CLI) or <suggest>true</suggest> (if using the WildFly Maven Plugin).
+To enable add-ons add the --add-ons=<list of add-ons> option to the scan command
+Some errors have been reported, you should fix them prior to provision a server with the --provision=SERVER option of the scan command
+----
+
```

Comment:
Maybe some output from running with --suggest. If so, I don't think we need ALL the previous output